services/core-api/app/api/mines/comments/resources/mine_comment.py

```diff
@@ -0,0 +1,82 @@
+import uuid
+from flask_restplus import Resource, reqparse, fields, inputs
+from flask import request, current_app
+from datetime import datetime
+from werkzeug.exceptions import BadRequest, NotFound, InternalServerError
+from sqlalchemy.exc import DBAPIError
+
+from app.extensions import api, db
+from app.api.utils.custom_reqparser import CustomReqparser
+from app.api.utils.resources_mixins import UserMixin
+from app.api.utils.access_decorators import requires_role_view_all, requires_role_mine_edit, requires_role_mine_admin
+
+from app.api.mines.mine.models.mine import Mine
+from app.api.mines.comments.models.mine_comment import MineComment
+
+from app.api.mines.response_models import MINE_COMMENT_MODEL
+
+
+class MineCommentListResource(Resource, UserMixin):
+    parser = CustomReqparser()
+    parser.add_argument('mine_comment', type=str, location='json')
+
+    @api.doc(description='retrive a list of comments for amine')
+    @api.marshal_with(MINE_COMMENT_MODEL, envelope='records', code=200)
+    @requires_role_view_all
+    def get(self, mine_guid):
+
+        mine = Mine.find_by_mine_guid(mine_guid)
+
+        if not mine:
+            raise NotFound('Mine not found')
+
+        current_app.logger.info(f'Retrieving comments for {mine}')
+
+        comments = [comment.__dict__ for comment in mine.comments]
+
+        return comments, 200
+
+    @api.expect(MINE_COMMENT_MODEL)
+    @api.doc(description='creates a new comment for the Mine')
+    @api.marshal_with(MINE_COMMENT_MODEL, code=201)
+    @requires_role_mine_edit
+    def post(self, mine_guid):
```

Comment:
I thought we wanted to use Marshmallow moving forward for new endpoints. We don't want to do this here?

Reply:
I chose to keep this the same as the original comments end point and not go with marshmallow as I am not 100% sure on its interaction with the mine object.